src/cargo/util/toml/mod.rs

```diff
@@ -1672,14 +1672,11 @@ fn detailed_dep_to_dependency<P: ResolveToPath + Clone>(
     kind: Option<DepKind>,
 ) -> CargoResult<Dependency> {
     if orig.version.is_none() && orig.path.is_none() && orig.git.is_none() {
-        let msg = format!(
-            "dependency ({}) specified without \
+        anyhow::bail!(
```

Comment:
Question: What's the criteria of turning this into

* a normal warning
* * a critical warning that bails out later
* a fatal error bailing out here

Reply:
> turning this into a normal warning

This was already a warning, so no change there

> a fatal error bailing out here

When we discussed #13629 as a team, we seemed in favor of a hard error as this was deemed a bug that people haven't been clamoring for us to continue to support.

> a critical warning that bails out later

I honestly haven't touched that system much.  I think those "recoverable errors", right?  As in, we continue moving forward, seeing other messages, and but fail after all of that?

While I would want to move more towards error recovery, this has some weird properties
- These only show up when `emit_warnings` is called which looks to only be compile commands and `cargo fetch`
- The overall value of error recovery seems relatively low (not much more is really going to be shown but instead people might see other unrelated warnings)

Reply:
Thanks for the clarification. I understand this being a fatal error as a team decision. The recovery error is something I am also thinking about, yet it doesn't apply to this situation.